2024/YoctoProjectDeveloperDay/practical-sec-tee.md

```diff
@@ -0,0 +1,994 @@
+- TEE API specififactions (1-2 slides)
+  + focus Global Platform API
+    - show that something liek this exist,s try to explain the purpose
+- OPTEE - about + history
+  + https://www.iwavesystems.com/product/op-tee-for-android-and-linux/
+  + architecture, tee-supplicant
+- optee - features overview (1 slide)
+- fTPM - autopromotion + example of trusted service  (1 slide)
+- optee secure storage (1 slide)
+  + https://optee.readthedocs.io/en/latest/architecture/secure_storage.html
+
+extras:
+- What is Trusted Execution Environment?
+  + Normal World vs Secure World
+    - Different architecture overview
+  + Root of trust
+  + TEE use cases
+    - Secure storage
+
+- TEE practice (Daniil Klimuk)
+    - Introduction
+    - What to integrate and why?
+    - Compilation from source vs vendor binaries.
+    - Linux/U-Boot and Yocto integration.
+    - OP-TEE Secure Storage.
+    - Experiments on Rockchip RK3566 SoC.
+    - Zarhus OS.
+- Outro and sources.
+
+???
+
+- Time for this slide: 30s
+- Idea/goal of this slide: Present entire presentation in a short way
+- What to say:
+    - TODO
+- Notes:
+    - TODO
+
+---
+
+# What is Trusted Execution Environment?
+
+.center[ <img src="/img/TEE_High_Lvl_Overview.svg" height="250px"> ]
+
+<br>
+
+_"A trusted execution environment (TEE) means a secure area which can guarantee
+the confidentiality and integrity of the code and data inside of this area._
+
+???
+
+- "from general information to specifics"
+- There's no agreed on firm definition and a lot of sources define their own
+- Definition taken from _"Building Secure Firmware, Armoring the Foundation of
+the Platform"_
+  + Will be listed in resources
+  + **Usually** refers to the possibility of TEE being only isolated via
+software which is not the case in some definitions.
+- Can be isolated via software
+- The term TEE was coined somewhere in 2000s
+- Used to be "secure area of the main processor" but the definition has
+broadened
+  + Can even be a separate device, more on that later
+- A device can have multiple TEEs nowadays
+- Used implementation is mostly dependant on underlying CPU architecture
+
+---
+
+# Common Ground in TEE definition
+
+### What it ensures:
+
+- **Data integrity**: preventing unauthorized entities from altering data when
+data is being processed.
+
+- **Data confidentiality**: Guarantees the authenticity of the executed code.
+
+- **Code integrity**: The confidentiality of its code, data and runtime states
+stored on a persistent memory.
+
+???
+
+- This is the part that is universally agreed on
+- This exact definition is from Confidential Computing Consortium
+  - A Linux Foundation Project which aims to accelerate the adoption of Trusted
+  Execution Environment (TEE) technologies and standards
+  - With Confidential Computing being the protection of data in use by
+  performing computation in a hardware-based, attested Trusted Execution
+  Environment.
+
+---
+
+# TEE locations
```

Comment:
```suggestion
# TEE architectures
```

Reply:
This is word for word from Building Secure Firmware but yes, architectures sounds more professional

Reply:
Any fix here then?